skrf/circuit.py

```diff
@@ -1515,16 +1515,29 @@ def invalide_to_reduce(cnx: list[tuple[Network, int]]) -> bool:
 
         connections = tmp_cnxs
 
+    # cache the connections that have been processed to avoid loop
+    processed_network_names: set[str] = set()
+
     # get the total number of network ports in the specified connection
     def calculate_ports(cnx: list[tuple[Network, int]]) -> int:
         if invalide_to_reduce(cnx):
             return -1
 
-        unique_networks = len(set(ntwk.name for ntwk, _ in cnx))
+        name_list = sorted(ntwk.name for ntwk, _ in cnx)
+        ntwks_str: str = ''.join(name_list)
+
+        unique_networks = len(set(name_list))
         total_ports = sum(ntwk.nports for ntwk, _ in cnx) - 2
 
         # Return the number of ports if the connections performed
-        return total_ports if unique_networks == 2 else total_ports // 2 - 1
+        ports = total_ports if unique_networks == 2 else total_ports // 2 - 1
+
+        if ntwks_str in processed_network_names:
+            ports -= 1
+        else:
+            processed_network_names.add(ntwks_str)
```

Comment:
it would worth adding a comment above explaning the logic behind

Reply:
Changes made as suggested.